lib/bolt/applicator.rb

```diff
@@ -199,9 +199,10 @@ def apply(args, apply_body, scope)
             @executor.with_node_logging("Applying manifest block", batch) do
               arguments = { 'catalog' => future.value, 'plugins' => plugins, '_noop' => options['_noop'] }
               raise future.reason if future.rejected?
-              result = transport.batch_task(batch, catalog_apply_task, arguments, options, &notify)
-              result = provide_puppet_missing_errors(result)
-              identify_resource_failures(result)
+              results = transport.batch_task(batch, catalog_apply_task, arguments, options, &notify)
+              Array(results).map do |result|
```

Comment:
I'm not clear why this change was necessary. Is the Orch transport returning a different format?

Reply:
The SSH and WinRM transports assume that their batch size is always 1, and they return a single result. The Orch transport supports actual batches, so it returns an array of results. We do similar munging [in the executor](https://github.com/puppetlabs/bolt/blob/master/lib/bolt/executor.rb#L78-L81)